gulpfile.js

```diff
@@ -51,6 +52,49 @@ const { launch, connect } = require('hadouken-js-adapter');
 		console.log(`[${new Date().toLocaleTimeString()}] ${chalk[color][bgcolor](msg)}.`);
 	}
 
+	/** 
+	* Validates the current node version against supported node versions specified in this file
+	* Returns boolean indicating whether current node version is valid
+	* Currently only validates against a max node version
+	*
+	* Note: This method is being used instead of npm engines because of an npm bug where warnings don't print
+	* This bug was resolved in npm 6.12.0 but as that is a very new version of npm and is not linked to node 10.15.3
+	* in nvm we can't assume our users have access to this version.
+	*/
+	const isNodeVersionValid = () => {
+		let ret = true;
+		// split the current node version into major, minor and patch numbers
+		let version = process.version.split('v');
+		version = version[1].split('.');		
+		const majorCurrent = Number(version[0]);
+		const minorCurrent = Number(version[1]);
+		const patchCurrent = Number(version[2]);
+
+		// split the max node version into major, minor and patch numbers
+		let maxNodeArray = MAX_NODE_VERSION.split('.');
+		if (maxNodeArray.length !== 3) {
+			console.log("Format of maximum node version must be: 'X.X.X', unable to validate node version");
+			return ret;
+		}
+		const majorMAX = Number(maxNodeArray[0]);
+		const minorMAX = Number(maxNodeArray[1]);
+		const patchMAX = Number(maxNodeArray[2]);
+
+		// major version is too high
+		if (majorCurrent > majorMAX ) {
+			ret = false;
+		} else if (majorCurrent === majorMAX) {
+			// major version matches the max allowed version, minor version is too high
+			if (minorCurrent > minorMAX) {
+				ret = false;
+			// major and minor version matches the max allowed version, patch version is too high
+			} else if (minorCurrent === minorMAX && patchCurrent > patchMAX) {
```

Comment:
You're returning false  on all. major, minor, patch. If that is intentional then you don't need that many if /else statements, it could be simple done with
```javascript
const max = MAX_NODE_VERSION.replace(/\D/, '')
const current = process.version.replace(/\D/, '')
if (current > max ) {
  return false
}
```
Of maybe I'm missing something?

Reply:
I think that would not not work as intended comparing
"v2.13.0"
to
"v2.2.0"

Reply:
Because of the length yeah, you have got `toFixed(6)`

Reply:
If you made each number a fixed length, then yes, that comparison would work.
Meredith and I came up with a different way that she is implementing, though.
basically it's

```
compareVersions(a, b)
  if (a.majorVersion !== b.majorVersion)
    return a.majorVersion < b.majorVersion ? -1 : 1;
  if (a.minorVersion !== b.minorVersion )
    return a.minorVersion < b.minorVersion ? -1 : 1;
  if (a.patchVersion !== b.patchVersion)
    return a.patchVersion< b.patchVersion? -1 : 1;
  return 0;
```

and then you use the comparison result to know whether the whole version `a` is less than the whole version `b`.

Reply:
I think that has too many `if` `if`

```javascript
const parse = (version) => {
  return `${version.replace(/\D/g, '') + '000'}`.substr(0, 6)
}

if (parse(current) > parse(max) ) {
  return false;
}

```

Reply:
@amaroks I've just checked in a refactor of the whole thing, so please take a look. 
I'd prefer not to have that many ifs either, but I've stared at your solution for several minutes and still have no idea how it works. I think it's okay to have a few extra lines to make it more readable.

Reply:
ok @meredith-ciq fair enough